app/components/retriable_fast_image/index.tsx

```diff
@@ -2,40 +2,86 @@
 // See LICENSE.txt for license information.
 
 import React, {PureComponent} from 'react';
+import CompassIcon from '@components/compass_icon';
 import FastImage, {FastImageProps} from 'react-native-fast-image';
+import validator from 'validator';
+import {StyleSheet, View} from 'react-native';
 
 export const FAST_IMAGE_MAX_RETRIES = 3;
 
-type RetriableFastImageProps = FastImageProps & {
-    id: string
+interface RetriableFastImageState {
+    retries: number,
 }
 
-type RetriableFastImageState = {
-    retry: number
+interface RetriableFastImageProps extends FastImageProps {
+    id: string
+    retry?: boolean
+    renderErrorImage?: boolean
+    errorImageSize?: number
 }
 
 export default class RetriableFastImage extends PureComponent<RetriableFastImageProps, RetriableFastImageState> {
     state = {
-        retry: 0,
+        retries: 0,
     }
 
+    config: validator.IsURLOptions = {
+        protocols: ['file', 'http', 'https', 'content', 'data'],
+        require_host: false,
+        require_tld: false,
+        require_valid_protocol: true,
+        require_protocol: true,
+        allow_underscores: true,
+        allow_trailing_dot: true,
+    };
+
+    // Checks if given URI is valid
+    isValidUri = this.props.source && typeof this.props.source != 'number' && this.props.source.uri ? validator.isURL(this.props.source.uri, this.config) || validator.isDataURI(this.props.source.uri) : false;
+
+    styles = StyleSheet.create({
+        iconView: {
+            flex: 1,
+            alignItems: 'center',
+            justifyContent: 'center',
+            alignContent: 'center',
+            height: '100%',
+        },
+    })
+
     onError = () => {
-        const retry = this.state.retry + 1;
-        if (retry > FAST_IMAGE_MAX_RETRIES && this.props.onError) {
-            this.props.onError();
+        const retryCount = this.state.retries + 1;
+        if (!this.isValidUri || retryCount > FAST_IMAGE_MAX_RETRIES) {
+            if (this.props.onError) {
+                this.props.onError();
+            }
             return;
         }
 
-        this.setState({retry});
+        this.setState({retries: retryCount});
     }
 
     render() {
-        return (
-            <FastImage
-                {...this.props}
-                key={`${this.props.id}-${this.state.retry}`}
-                onError={this.onError}
-            />
-        );
+        if (this.isValidUri) {
```

Comment:
retryCount should be considered here, right ?

Reply:
Possibly? There's two error handling cases happening though -

1. `<FastImage onError={this.onError()}` which essentially triggers `this.props.onError` - and that's for the scenario where the `FastImage` component fails
2. `isValidUri` which just displays the `<CompassIcon name='jumbo-attachment-image-broken' ...` even before `FastImage` is called, provided that prop is set